components/match2/wikis/mobilelegends/match_summary.lua

```diff
@@ -0,0 +1,328 @@
+		end
+
+		body:addRow(championBan)
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createGame(game, gameIndex)
+	local row = MatchSummary.Row()
+	local extradata = game.extradata or {}
+
+	local championsData = {{}, {}}
+	local championsDataIsEmpty = true
+	for champIndex = 1, _NUM_CHAMPIONS_PICK do
+		if String.isNotEmpty(extradata['team1champion' .. champIndex]) then
+			championsData[1][champIndex] = extradata['team1champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		if String.isNotEmpty(extradata['team2champion' .. champIndex]) then
+			championsData[2][champIndex] = extradata['team2champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		championsData[1].color = extradata.team1side
+		championsData[2].color = extradata.team2side
+	end
+
+	if
+		String.isEmpty(game.length) and
+		String.isEmpty(game.winner) and
+		championsDataIsEmpty
+	then
+		return nil
+	end
+
+	row	:addClass('brkts-popup-body-game')
+		:css('font-size', '85%')
+		:css('overflow', 'hidden')
+
+	row:addElement(CustomMatchSummary._opponentChampionsDisplay(championsData[1], _NUM_CHAMPIONS_PICK, false))
+	row:addElement(CustomMatchSummary._createCheckMark(game.winner == 1))
+	row:addElement(mw.html.create('div')
+		:addClass('brkts-popup-body-element-vertical-centered')
+		:wikitext(CustomMatchSummary._createAbbreviation{
+			title = String.isEmpty(game.length) and ('Game ' .. gameIndex .. ' picks') or 'Match Length',
+			text = game.length or ('Game ' .. gameIndex),
+		})
+	)
+	row:addElement(CustomMatchSummary._createCheckMark(game.winner == 2))
+	row:addElement(CustomMatchSummary._opponentChampionsDisplay(championsData[2], _NUM_CHAMPIONS_PICK, true))
+
+	-- Add Comment
+	if not Logic.isEmpty(game.comment) then
+		row:addElement(MatchSummary.Break():create())
+		local comment = mw.html.create('div')
+		comment :wikitext(game.comment)
+				:css('margin', 'auto')
+		row:addElement(comment)
+	end
+
+	return row
+end
+
+function CustomMatchSummary._createCheckMark(isWinner)
+	local container = mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+		:css('line-height', '27px')
+		:css('margin-left', '3%')
+		:css('margin-right', '3%')
+
+	if isWinner then
+		container:node(_GREEN_CHECK)
+	else
+		container:node(_NO_CHECK)
+	end
+
+	return container
+end
+
+function CustomMatchSummary._createOpponent(opponent, side)
+	return OpponentDisplay.BlockOpponent{
+		flip = side == 'left',
+		opponent = opponent,
+		overflow = 'wrap',
+		teamStyle = 'short',
+	}
+end
+
+function CustomMatchSummary._createScore(opponent)
+	return OpponentDisplay.BlockScore{
+		isWinner = opponent.placement == 1 or opponent.advances,
+		scoreText = OpponentDisplay.InlineScore(opponent),
+	}
+end
+
+function CustomMatchSummary._createAbbreviation(args)
+	return '<i><abbr title="' .. args.title .. '">' .. args.text .. '</abbr></i>'
+end
+
+function CustomMatchSummary._opponentChampionsDisplay(opponentChampionsData, numberOfChampions, flip, isBan)
```

Comment:
is `numberOfChampions` needed? Why can't `#opponentChampionsData` and/or `ipairs(opponentChampionsData)` be used?

Reply:
there should be same number of champion bans for both teams
if one is not entered it is left nil
the icon module returns a transparent icon in case of nil